### ISSUE TITLE
problem with semidirect product construction

### ISSUE THREAD

**Issue by ThomasBreuer** [gap-system/gap#5266]
In GAP 4.12.1 and in the current master branch, we get the following.
```
gap> p:= 5;;
gap> z:= Z(p^2);;
gap> N:= Group( z );;   # the multiplicative group of the field with 25 elements
gap> alpha:= GroupHomomorphismByImages( N, N, [ z ], [ z^p ] );;
gap> G:= Group( alpha );;   # a group that acts on N
gap> SemidirectProduct( G, N );
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 1st choice method found for `SemidirectProduct' on 3 arguments at
[...]
```
The point is that the relevant `SemidirectProduct` method is applicable only if the group `G` stores that it is finite.
Thus we do not run into an error if we proceed as follows.
```
gap> IsFinite( G );
true
gap> SemidirectProduct( G, N );
<pc group with 5 generators>
```
Can't we simply remove the `IsFinite` requirement from the method installation, and check for finiteness inside the method?

**Comment by hulpke** [comment]
The reason for the `IsFinite` is that the method used calculates a faithful permutation and pc representation of the group.
(One could put a test 
```
if not IsFinite(G) then TryNextMethod();fi;
```
in the function instead -- in fact there will be lots of such situations for other operations.)

As far as I know we do not have a generic `SemidirectProduct` method that represents elements as pairs.

**Comment by ThomasBreuer** [comment]
@ahulpke Yes, I had thought about this change.

**Comment by ThomasBreuer** [comment]
After #5267 got merged, the original problem is fixed.
I am not sure whether the issue should be closed, because of the remarks of @hulpke that
1. there are (a lot of) similar situations in the GAP library which could be improved in a similar way and
2. there are apparently no generic semidirect products, contrary to the situations with direct products and wreath products.

Should these comments better be addressed by separate issues?

**Comment by hulpke** [comment]
I think both items are easily separate issues of their own.

**Comment by fingolfin** [comment]
For 2. from above @ThomasBreuer open a separate issue #5272. And 1. is not actionable in that form -- i.e., "something somewhere is broken" is not something we can handle. Instead we'll handle this as we find them, or if someone else reports them.
